src/DynamoCore/Library/XmlDocumentationExtensions.cs

```diff
@@ -199,6 +222,81 @@ private static string GetMemberElementName(FunctionDescriptor member)
             return String.Format("{0}:{1}", prefixCode, memberName);
         }
 
+        private enum XmlTagType
+        {
+            None,
+            Member,
+            Summary,
+            Parameter,
+            SearchTags
+        }
+
+        private static void LoadDataFromXml(XmlReader reader)
+        {
+            if (reader == null)
+                return;
+
+            MemberDocumentNode currentDocNode = new MemberDocumentNode();
+            XmlTagType currentTag = XmlTagType.None;
+            string currentParamName = String.Empty;
+
+            while (reader.Read())
+            {
+                switch (reader.NodeType)
+                {
+                    case XmlNodeType.Element:
+                        switch (reader.Name)
+                        {
+                            case "member":
+                                // Find attribute "name".
+                                if (reader.MoveToAttribute("name"))
+                                {
+                                    currentDocNode = new MemberDocumentNode(reader.Value);
+                                    documentNodes.Add(currentDocNode.FullyQualifiedName, currentDocNode);
+                                }
+                                currentTag = XmlTagType.Member;
+                                break;
+
+                            case "summary":
+                                currentTag = XmlTagType.Summary;
+                                break;
+
+                            case "param":
+                                if (reader.MoveToAttribute("name"))
+                                {
+                                    currentParamName = reader.Value;
+                                }
+                                currentTag = XmlTagType.Parameter;
+                                break;
+
+                            case "search":
+                                currentTag = XmlTagType.SearchTags;
+                                break;
+
+                            default:
+                                currentTag = XmlTagType.None;
+                                break;
+                        }
+                        break;
+                    case XmlNodeType.Text:
+                        switch (currentTag)
+                        {
+                            case XmlTagType.Summary:
+                                currentDocNode.Summary = reader.Value;
```

Comment:
If you put a breakpoint here, you'll see `reader.Value` returns string contents with `\n` and spaces. Do you think we need to trim those? If we do, then perhaps we can do it inside `Summary` property setter.